regression-tests/mixed-function-type-argument.cpp2

```diff
@@ -0,0 +1,18 @@
+#include <cstddef>
+main: () = {
+  _ = alignof(int);
+  _ = sizeof(int);
+  _ = typeid(int);
+  _ = offsetof(t, a);
+  _ = :<T> (_: T) _ = sizeof(T::value);(std::true_type());
+  _ = :<T> (_: T) _ = sizeof(type T::value_type);(std::true_type());
```

Comment:
Is there intended to be a difference between lines 7 and 8? Or is the `type` intended to be optional?

If there's no difference, can you show a case where it's required?

Reply:
The second one needs the disambiguating `type`.
See <https://cpp1.godbolt.org/z/PY53n6Kvf> (generated from <https://cpp2.godbolt.org/z/KK15vf6Y5>):
- Clang: `error: dependent-name 'T::value_type' is parsed as a non-type, but instantiation yields a type`.
- GCC: `error: missing 'typename' prior to dependent type name 'integral_constant<bool, true>::value_type'`.
- MSVC accepts.

Reply:
D'oh, now I see it. I missed the `_type` suffix that changed the meaning, sorry.